blog/content/post/2020-08-26-coreboot-riscv.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Building coreboot for RISC-V and running it in Qemu
+abstract: 'In this article, I will show you step by step how to build coreboot for RISC-V and run it in QEMU emulator'
+cover: /covers/coreboot-logo.svg
+author: wojciech.niewiadomski
+layout: post
+published: true
+date: 2020-08-26
+archives: "2020"
+
+tags:
+  - Firmware
+  - coreboot
+  - RISC-V
+categories:
+  - Firmware
+  - Miscellaneous
+
+---
+
+Your post content
+
+# Building coreboot for RISC-V and running it in Qemu
+
+#### 1. Install required tools and libraries
+
+On Debian:
+
+```sh
+$ sudo apt-get install -y bison build-essential curl flex git gnat libncurses5-dev m4 zlib1g-dev qemu-system
+```
+On Arch Linux:
+
+```sh
+$ sudo pacman -S base-devel curl git gcc-ada ncurses zlib
+```
+
+On Fedora:
+
+```sh
+$ sudo dnf install git make gcc-gnat flex bison xz bzip2 gcc g++ ncurses-devel wget zlib-devel
+```
+
+#### 2. Download coreboot source tree
+
+```sh
+$ git clone https://review.coreboot.org/coreboot
+$ cd coreboot
+```
+
+#### 3. Build coreboot toolchain
+
+You can use CPUS= to specify the number of threads you want to use for building
+toolchain.
+
+```sh
+$ make crossgcc-riscv CPUS=4
+```
+> NOTE: This will take a while
+
+At the end of the process, you should get the info that your cross toolchain is
+built
+
+```
+You can now run riscv64-elf cross GCC from /your/path/to/coreboot/util/crossgcc/xgcc.
+```
+
+#### 4. Configure payload
```

Comment:
This step is obsolete. coreboot integrates payloads automatically. And I'm not sure SeaBIOS is supported on RISC-V